presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -146,15 +127,12 @@ public NodeSelector createNodeSelector(String dataSourceName)
 
             for (Node node : nodes) {
                 if (useNetworkTopology && (includeCoordinator || !coordinatorNodeIds.contains(node.getNodeIdentifier()))) {
-                    try {
-                        NetworkLocation location = networkLocationCache.get(node.getHostAndPort());
-                        for (int i = 0; i <= location.getSegments().size(); i++) {
-                            workersByNetworkPath.put(location.subLocation(0, i), node);
+                    Optional<NetworkLocation> location = networkLocationCache.get(node.getHostAndPort());
+                    if (location.isPresent()) {
+                        for (int i = 0; i <= location.get().getSegments().size(); i++) {
+                            workersByNetworkPath.put(location.get().subLocation(0, i), node);
                         }
                     }
```

Comment:
You need to handle the else case, now that it's expected that the network location could be empty before the cache is loaded.
```
else {
 // just add it to the root location, if we couldn't locate the worker
 workersByNetworkPath.put(new NetworkLocation(), node);
}
```